src/main/java/com/mark0wka/controllers/CinemaController.java

```diff
@@ -18,74 +18,138 @@
 public class CinemaController {
 
     @Autowired
-    CinemaObjectRepository ObjRepository;
+    CinemaObjectRepository objectRepository;
 
     @Autowired
-    AttributeRepository AttrRepository;
+    AttributeRepository attributeRepository;
 
     @GetMapping("/cinema/all")
     public List<EntityObject> getAllCinemas() {
-        return ObjRepository.findAllByObjectTypeId("Cinema");
+        return objectRepository.findAllByObjectTypeId("Cinema");
     }
 
     @GetMapping("/films/all")
     public List<EntityObject> getAllFilms() {
-        return ObjRepository.findAllByObjectTypeId("Film");
+        return objectRepository.findAllByObjectTypeId("Film");
     }
 
-    @PostMapping("/cinema")
-    public Cinema createCinema(@RequestBody Cinema cinema) {
-        System.out.println(cinema);
-        EntityObject obj = cinema.getEntity();
-        System.out.println(obj);
-        ObjRepository.save(cinema.getEntity());
-        //AttributeRepository.save
-        return cinema;
+    @GetMapping("/halls/all")
+    public List<EntityObject> getAllHalls() {
+        return objectRepository.findAllByObjectTypeId("Hall");
+    }
+
+    @GetMapping("/seats/all")
+    public List<EntityObject> getAllSeats() {
+        return objectRepository.findAllByObjectTypeId("Seat");
+    }
+
+    @GetMapping("/sessions/all")
+    public List<EntityObject> getAllSessions() {
+        return objectRepository.findAllByObjectTypeId("Session");
+    }
+
+    @GetMapping("/halls/{cinemaName}")
+    public Set<EntityObject> getHallsByCinemaName(@PathVariable String cinemaName) {
+        Set<EntityObject> result = new HashSet<>();
+        List<Attribute> id = attributeRepository.findByAttrIdAndValue(2, String.valueOf(objectRepository.findEntityObjectByName(cinemaName).getObjectId()));
+        id.forEach(x -> result.add(objectRepository.findEntityObjectByObjectId(x.getObjectId())));
```

Comment:
Let's do it in a functional way like, its more readable
<pre>ids.stream()
    .map(Attribute::getObjectId)
    .map(CinemaObjectRepository::findEntityObjByObjId)
    .collect(Collectors.toSet())</pre>
You see it? one row - one transformation